clients/roscpp/src/libros/service_server_link.cpp

```diff
@@ -321,7 +322,7 @@ void ServiceServerLink::processNextCall()
   }
 }
 
-bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& resp)
+bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& resp, double timeout)
```

Comment:
Following my previous comment above, here you'll have to do this:

``` c++
``` c++
bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& resp)
{
  return call(req, resp, -1);
}

bool ServiceServerLink::call(const SerializedMessage& req, SerializedMessage& resp, double timeout)
```

Reply:
Thanks for notice. Fixed